scripts/nb-tester/qiskit_docs_notebook_tester/execute.py

```diff
@@ -130,11 +131,12 @@ async def _execute_notebook(
     """
     nb = nbformat.read(job.path, as_version=4)
 
-    kernel_manager, kernel = await start_new_async_kernel(
-        kernel_name="python3",
-        extra_arguments=["--InlineBackend.figure_format='svg'"],
-        cwd=working_directory,
-    )
+    async with kernel_setup_lock:
```

Comment:
It'd be helpful to add a comment, e.g.

> We've noticed non-deterministic failures when running multiple notebooks in parallel. We believe the issue is when starting up new kernels, whereas it's okay to have multiple already created kernels continue their execution in parallel.

Probably look to the issue you got inspiration from